pySDC/implementations/controller_classes/controller_nonMPI.py

```diff
@@ -712,3 +746,61 @@ def default(self, local_MS_running):
             local_MS_running (list): list of currently running steps
         """
         raise ControllerError('Unknown stage, got %s' % local_MS_running[0].status.stage)  # TODO
+
+    def resilence(self, local_MS_running):
+        """
+        Call various functions that are supposed to provide some sort of resilience from here
+        """
+
+        if self.params.use_HotRod:
+            self.hotrod(local_MS_running)
+
+    def hotrod(self, local_MS_running):
+        """
+        See for the reference:
+        Lightweight and Accurate Silent Data Corruption Detection in Ordinary Differential Equation Solvers,
+        Guhur et al. 2016, Springer. DOI: 10.1007/978-3-319-43659-3_47
+        """
+        for i in range(len(local_MS_running)):
+            S = local_MS_running[i]
+            if S.status.iter == S.params.maxiter:
+                for l in S.levels:
+                    # throw away the final sweep to match the error estimates
+                    l.u[:] = l.uold[:]
+
+                    # check if a fault is detected
+                    if None not in [l.status.e_extrapolated, l.status.e_embedded]:
+                        diff = l.status.e_extrapolated - l.status.e_embedded
```

Comment:
Can you/do you make sure these values are set before using them? And set to something meaningful?